replica/database.cc

```diff
@@ -361,7 +361,7 @@ database::database(const db::config& cfg, database_config dbcfg, service::migrat
     , _feat(feat)
     , _shared_token_metadata(stm)
     , _sst_dir_semaphore(sst_dir_sem)
-    , _wasm_engine(std::make_unique<wasm::engine>())
+    , _wasm_engine(std::make_unique<rust::Box<wasmtime::Engine>>(wasmtime::create_engine()))
```

Comment:
Isn't `rust::Box` something like a `std::unique_ptr` already? Would it be possible to just use `rust::Box` directly and remove the double indirection?

Reply:
I tried using the `rust::Box` alone in the rebase and looks like it works for now